module-1/src/Components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,78 @@
+import { ReactNode } from "react";
+import { createPortal } from "react-dom";
+import "./Dialog.css";
+const mountElement: any = document.getElementById("overlays") || <div />;
+
+interface IDialogProps {
+  openDialog: boolean;
+  dialogTitle?: string;
+  children?: ReactNode;
+  closeCallback: () => void;
+  submitCallback: () => void;
+}
+
+export const Dialog = ({
+  openDialog = false,
+  dialogTitle,
+  children,
+  closeCallback,
+  submitCallback,
+}: IDialogProps) => {
+  if (!openDialog) return null;
+  function handleClose(): void {
+    closeCallback();
+  }
+
+  function handleSubmit(e: React.FormEvent<HTMLFormElement>): void {
+    e.preventDefault();
+    submitCallback();
+  }
+  return createPortal(
+    <>
+      <div className="fixed left-0 top-0 bg-black bg-opacity-5 w-screen h-screen"></div>
+      <div className="dialog flex flex-col justify-between bg-custom_dark_gray min-w-40 min-h-32">
+        <div id="dialogHeader">
+          <div className="flex text-white justify-end mb-1">
+            <svg
+              xmlns="http://www.w3.org/2000/svg"
+              fill="none"
+              viewBox="0 0 24 24"
+              strokeWidth={1.5}
+              stroke="currentColor"
+              className="w-6 h-6"
+              onClick={() => handleClose()}
+            >
+              <path
+                strokeLinecap="round"
+                strokeLinejoin="round"
+                d="M6 18 18 6M6 6l12 12"
+              />
+            </svg>
+          </div>
+          <h1 className="text-white font-Montserrat ml-2 mb-2">
+            {dialogTitle}
+          </h1>
+        </div>
+        <form onSubmit={(e) => handleSubmit(e)}>
```

Comment:
Because `Dialog` are very generic you must keep it as feature-agnostic as possible. `form` and submit/cancel buttons are logically a part of the form and should be moved there.

Reply:
updated